### ISSUE TITLE
Custom Dataset Training problem after few Epochs

### ISSUE THREAD

**Issue by komms** [CSAILVision/semantic-segmentation-pytorch#168]
I deleted some of the labels based on my use-case and while training in the 3 epoch, I got an assertion error as shown in the figure.  
![Screenshot from 2019-05-15 13-55-03](https://user-images.githubusercontent.com/35064612/57773908-6beb9600-7719-11e9-9687-94323d125ac3.png)
![Screenshot from 2019-05-15 13-58-08](https://user-images.githubusercontent.com/35064612/57773948-86be0a80-7719-11e9-84c4-9736ae662c9b.png)

The network has already processed the whole training images twice, according to batch size and no of iterations. I don't understand it showing an error that the image doesn't have 3 dimensions. Can anyone provide me an insight?
Thanks in advance :)
@hangzhaomit  @Tete-Xiao

**Comment by komms** [comment]
The problem was with dataset as clearly stated in the error.
